Tests/SwiftCliCoreTests/OffsetUtilityTest.swift

```diff
@@ -0,0 +1,169 @@
+    }
+
+    func testFindRightDecForOddRow() throws {
+        let result = OffsetUtility().findRightChanges(stitchRow: ["k2tog", "k1", "k1", "k2tog", "k1", "p1", "k2tog"])
+        let expectedResult = -2
+        expect(result).to(equal(expectedResult))
+    }
+
+    func testFindRightIncForOddRow() throws {
+        let result = OffsetUtility().findRightChanges(stitchRow: ["k1", "yo", "k1", "p1", "k1", "p1", "yo", "k1"])
+        let expectedResult = 1
+        expect(result).to(equal(expectedResult))
+    }
+    func testFindRightIncForEvenRow() throws {
+        let result = OffsetUtility().findRightChanges(stitchRow: ["k1", "k1", "p1", "k1", "yo", "k1", "p1", "k1"])
+        let expectedResult = 1
+        expect(result).to(equal(expectedResult))
+    }
+
+}
+
+class MakeRowMetadataTests: XCTestCase {
+
+    func testFindRightDecForManyStith() throws {
+        let result = OffsetUtility().makeRowMetadata(stitchRow: ["k1", "yo", "k1", "p1", "k1", "p1", "yo", "k1"], rowNumber: 0)
+        let expectedResult = RowInfo(row: ["k1", "yo", "k1", "p1", "k1", "p1", "yo", "k1"],
+                                     rowNumber: 0,
+                                     bottomLine: "└─┴─┴─┴─┴─┴─┴─┴─┘",
+                                     stitchSymbols: "│ │o│ │-│ │-│o│ │\n",
+                                     width: 8,
+                                     leftIncDec: 1,
+                                     rightIncDec: 1,
+                                     leftOffset: 0)
+        expect(result).to(equal(expectedResult))
+    }
+
+}
+
+class gatherAllMetaDataTests: XCTestCase {
+
+    func testGatherAllMetaDataTests() throws {
+        let result = OffsetUtility().gatherAllMetaData(stitchArray: [["k1", "p1"], ["k1", "p1"]])
+
+        let expectedResult = [RowInfo(row: ["k1", "p1"],
+                                     rowNumber: 0,
+                                     bottomLine: "└─┴─┘",
+                                     stitchSymbols: "│ │-│\n",
+                                     width: 2,
+                                     leftIncDec: 0,
+                                     rightIncDec: 0,
+                                     leftOffset: 0),
+                              RowInfo(row: ["k1", "p1"],
+                                       rowNumber: 1,
+                                       bottomLine: "├─┼─┤\n",
+                                       stitchSymbols: "│ │-│\n",
+                                       width: 2,
+                                       leftIncDec: 0,
+                                       rightIncDec: 0,
+                                       leftOffset: 0)]
+        expect(result).to(equal(expectedResult))
+    }
+
+    func testGatherAllMetaDataTestsLeftInc() throws {
+        let result = OffsetUtility().gatherAllMetaData(stitchArray: [["k1", "k1", "p1"], ["k1", "yo", "k1", "p1"]])
+
+        let expectedResult = [RowInfo(row: ["k1", "k1", "p1"],
```

Comment:
When you can't fit all the parameters on a single line, start the first parameter on a new line, e.g.

```swift
RowInfo(
  row: ["k1", "k1", "p1"], 
  rowNumber: 0,
  ...
)
```